test/Fencer/Server/Test.hs

```diff
@@ -133,17 +231,39 @@ withServer
 withServer =
   withResource createServer destroyServer
 
+-- | Get a unique port number that is not used by other tests. Needed since
+-- all tests run in parallel.
+getUniquePort :: IO Port
+getUniquePort = atomicModifyIORef' next (\port -> (succ port, port))
+  where
+    next :: IORef Port
+    next = unsafePerformIO (newIORef defaultGRPCPort)
+    {-# NOINLINE next #-}
```

Comment:
I'm new to these pragmas for compiler. What is this one for, i.e., why is it important to have it?

Reply:
https://github.com/ndmitchell/blogs/blob/master/published/hlint-unsafeperformio.md

Reply:
Ok, got it!